src/main/scala/com/advancedtelematic/campaigner/db/Campaigns.scala

```diff
@@ -117,13 +125,29 @@ protected [db] class Campaigns(implicit db: Database, ec: ExecutionContext)
   def findCampaignsByUpdate(update: UpdateId): Future[Seq[Campaign]] =
     db.run(campaignRepo.findByUpdateAction(update))
 
+  /**
+   * Calculate campaign-wide statistic counters, also taking retry campaings
+   * into account if any exist.
+   */
   def campaignStats(campaignId: CampaignId): Future[CampaignStats] = for {
-    status    <- findClientCampaign(campaignId).map(_.status)
-    finished  <- countFinished(campaignId)
-    cancelled <- countCancelled(campaignId)
-    failed    <- failedDevices(campaignId)
-    stats     <- campaignStatsFor(campaignId)
-  } yield CampaignStats(campaignId, status, finished, failed, cancelled, stats)
+    mainCampaign <- campaignRepo.find(campaignId)
+    retryCampaignIds <- campaignRepo.findChildIdsOf(campaignId)
+    Stats(mainProcessed, mainAffected) <- campaignStatsFor(campaignId)
+    Stats(retryProcessed, retryAffected) <- campaignStatsForAllOf(retryCampaignIds)
+    mainCancelled <- countCancelled(Set(campaignId))
```

Comment:
The `Stats` object is coming from group stats, while `cancelled` is calculated using device stats. They could be put together, but that would require a lot of refactoring, which I tried to avoid for this PR. Looking at your later comment, though, it doesn't seem to be such a bad idea now. I'll fix the code to be executed in one transaction and see what I can do for this issue at the same time.